tests/nightly/test_large_array.py

```diff
@@ -217,6 +260,33 @@ def numpy_space_to_depth(x, blocksize):
     output = mx.nd.space_to_depth(data, 2)
     assert_almost_equal(output.asnumpy(), expected, atol=1e-3, rtol=1e-3)
 
+
+def test_diag():
+    h = np.random.randint(2,9)
```

Comment:
why use randint? Can we just use deterministic values to test diag?

Reply:
@apeforest : using random values is better since we are checking diagonal. Having different values actually ensures that the operation is performed correctly. Checking just the shape is fine but this ensures total correctness. I think if we can perform this check then it ensures total correctness